src/formatDate.js

```diff
@@ -50,7 +50,93 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const NOW_CENTURY = '20';
+  const LAST_CENTURY = '19';
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const newArray = [];
+  let indexYear;
+  let day;
+  let month;
+  let year;
+
+  function getValue(strDate, Separator, index) {
+    const indexOfSeparator1 = date.indexOf(Separator);
+    const indexOfSeparator2 = date.indexOf(Separator, indexOfSeparator1 + 1);
+
+    switch (index) {
+      case 0:
+        return strDate.slice(0, indexOfSeparator1);
+
+      case 1:
+        return strDate.slice(indexOfSeparator1 + 1, indexOfSeparator2);
+
+      case 2:
+        return strDate.slice(indexOfSeparator2 + 1);
+    }
+  }
+
+  for (const item of fromFormat) {
+    switch (item) {
+      case 'DD':
```

Comment:
DD, MM and YY are magic words
Consider will be better if you create a constants for magic words and use it everywhere